src/js/DropContainer.jsx

```diff
@@ -0,0 +1,284 @@
+      // user's file format input
+      isFileValid = this.validateFileFormat(
+         Array.prototype.slice.call(dataTransferItems), fileFormat);
+
+      this.setState({
+         isDragActive: isFileValid,
+         dropzoneStyle: 'dragBorder',
+         visibility: 'hide'
+      });
+
+      // DropContainer turns green if valid format
+      // Otherwise turns red for warning
+      if (isFileValid) {
+         this.correctFormatBorderStyle()
+      } else {
+         this.incorrectFormatBorderStyle();
+      }
+   }
+
+   /**
+      * Handle the drag over event
+      * @param  {Event]} event
+   */
+   onDragOver (event) {
+      event.preventDefault();
+      event.stopPropagation();
+      event.dataTransfer.dropEffect = 'copy';
+      return false;
+   }
+
+   /**
+      * Handle the drag leave event
+   */
+   onDragLeave (event) {
+      event.preventDefault();
+
+      // Only deactivate once the dropzone and all children was left.
+      if (--this.enterCounter > 0) {
+         return;
+      }
+
+      this.setState({
+         isDragActive: false,
+         isDragReject: false,
+         dropzoneStyle: '',
+         visibility: 'show'
+      });
+
+      // if (this.props.onDragLeave) {
+      //    this.props.onDragLeave.call(this, event);
+      // }
+
+      if(this.state.imageFile){
+         return;
+      }
+  }
+
+   /**
+      * Handle the drop event
+      * @param  {Event]} event
+   */
+   onDrop (event) {
+      event.preventDefault();
+      this.setState({
+         isDragActive: false,
+         dropzoneStyle: '',
+         visibility: 'show'
+      });
+      // Call handleChange function
+      this.handleChange.call(this, event);
+   }
+
+   /**
+      * Handle Input Button Click
+      * @param  {Event]} event
+   */
+   handleInputButtonClick (event) {
+      this.handleChange(event);
+   }
+
+   /**
+      * Handle the file state
+      * If file is valid => create a URL obj
+      * And modify the state value so the
+      * parent (UploadContainer) can access it
+      * @param  {Event]} event
+   */
+   handleChange (event) {
+      let e = event.nativeEvent,
+         dataTransferItems = this.extractData(event),
+         fileFormat = this.fileFormatFinder(this.props.accept),
+         isFileValid = this.validateFileFormat(Array.prototype.slice.call(dataTransferItems), fileFormat);
+
+      // Check to make sure the dropped file
+      // is in correct format - e.g image/*
+      if(isFileValid){
+         let eventFile = e.dataTransfer || event.target;
+
+         if(eventFile) {
+            this.dragAndDropFileUrlGenerator(eventFile)
+         } else {
+            this.inputButtonFileUrlGenerator(eventFile);
+         }
+      }
+   }
+
+   dragAndDropFileUrlGenerator (eventFile) {
+      let file = eventFile.files[0], uploadFile;
+
+      if(typeof file !== undefined) {
+         uploadFile = URL.createObjectURL(file);
+
+         this.setState({
+            fileSrc: uploadFile
```

Comment:
Let's renamed fileSrc, fileSrc sounds like a url when it's actually the file object.